src/System.Net.Http/src/netcore50/System/Net/HttpClientHandler.cs

```diff
@@ -321,7 +321,10 @@ public long MaxRequestContentBufferSize
 
         public int MaxResponseHeadersLength
         {
-            get { return 0; } // TODO: Issue #8541 - return default value used in Wininet.
+            // Windows.Web.Http is built on WinINet. There is no maximum limit (except for out of memory)
+            // for received response headers. So, returning 0 (indicating no limit) is appropriate.
+            get { return 0; }
```

Comment:
Good point. This actually was discussed a while back with @ericstj about what to do with properties that are "limits" and what should mean "infinite" or "no limit".  For TimeSpan that was easy since we had things like Timeout.InfiniteTimeSpan.

We are inconsistent about this actually. In the default .NET Framework, we allow 0 for some things to mean no limit.  
https://msdn.microsoft.com/en-us/library/system.net.http.httpclienthandler.maxrequestcontentbuffersize(v=vs.118).aspx
(Although this is not fully documented)

Using -1 is a very Win32 API kind of thing.  This property is not nullable; if it were, then we would set it to null.

I'm open to revisiting this discussion.  Also, we could tweak WinHttpHandler to allow for 0 as well to mean the same thing as infinite. In fact, many Win32 WinHTTP things are now allowing 0 and -1 to mean the same thing: "infinite" / "no limit".